aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -87,6 +87,13 @@ Tensor isnan(const Tensor& self) {
   return self != self;
 }
 
+Tensor isfinite(const Tensor& self) {
+  if (!self.is_floating_point()) {
+    return at::ones_like(self, at::kBool);
+  }
+  return (self == self) * (self.abs() != at::full_like(self, std::numeric_limits<double>::infinity()));
```

Comment:
It's faster to just compare with `at::full(1, std::numeric_limits<double>::infinity()))`
Also, is double infinity the same thing as float infinity?

Reply:
done